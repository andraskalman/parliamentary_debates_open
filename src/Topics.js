import React from 'react';
import { Checkbox } from 'react-bootstrap';

const Topics = ({ topics, onChange, counts }) => (
  <ul>
    {Object.keys(topics)
      .filter(topic => counts[topic] >= 1)
      .sort((a, b) => counts[a] <= counts[b])
      .map(topic => (
      <li key={topic}>
        <Checkbox onChange={event => onChange({
          topic,
          value: event.target.checked}
        )} checked={topics[topic]}>
          {topic} <b>({counts[topic]})</b>
        </Checkbox>
      </li>
    ))}
  </ul>
)

export default Topics
