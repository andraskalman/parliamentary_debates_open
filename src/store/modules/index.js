import { combineReducers } from 'redux'
import { routerReducer } from 'react-router-redux'
import search from './search'
import help from './help'

export default combineReducers({
  routing: routerReducer,
  search,
  help
})
