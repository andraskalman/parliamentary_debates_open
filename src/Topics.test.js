import React from 'react';
import ReactDOM from 'react-dom';
import Topics from './Topics';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'
import store, { history } from './store'
import { ConnectedRouter } from 'react-router-redux'


it('renders correctly', () => {
  const tree = renderer
    .create(<Topics
      topics={{"foo": true, "bar": false}}
      counts={{"foo": 3, "bar": 4}}
      />)
    .toJSON();
  expect(tree).toMatchSnapshot();
})
