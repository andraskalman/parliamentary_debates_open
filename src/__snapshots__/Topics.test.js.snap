// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul>
  <li>
    <div
      className="checkbox"
      style={undefined}
    >
      <label
        title=""
      >
        <input
          checked={false}
          disabled={false}
          onChange={[Function]}
          type="checkbox"
        />
        bar
         
        <b>
          (
          4
          )
        </b>
      </label>
    </div>
  </li>
  <li>
    <div
      className="checkbox"
      style={undefined}
    >
      <label
        title=""
      >
        <input
          checked={true}
          disabled={false}
          onChange={[Function]}
          type="checkbox"
        />
        foo
         
        <b>
          (
          3
          )
        </b>
      </label>
    </div>
  </li>
</ul>
`;
